firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/Containers/RingBuffer.txt

```diff
@@ -0,0 +1,592 @@
+// * indicates the element addressed by oldest_index_
+// ^ indicates the element addressed by newest_index_
+
+SCENARIO('Non-Volatile RingBuffer works correctly when oldest index is 0') 
+  GIVEN('A RingBuffer with capacity 4 to which 0 bytes of data were pushed, 0 bytes were poped') 
+    // internal state should look like [*^] [] [] [] []
+    WHEN('1 byte of data is pushed ') 
+      THEN('The push method returns ok status') 
+      THEN('After push method ,the peek method returns expected data ') 
+        ////internal state should look like [0*] [^] [] [][]
+
+    WHEN('1 byte of data is poped ') 
+      THEN('The pop method returns empty status')
+        // internal state should look like [*^] [] [] [][]
```

Comment:
We need a THEN to specify what the internal state should look like, e.g. "THEN the peek method returns empty on all indices"